test/parallel/test-child-process-spawnsync-validation-errors.js

```diff
@@ -0,0 +1,176 @@
+    fail('uid', true, err);
+    fail('uid', false, err);
+    fail('uid', [], err);
+    fail('uid', {}, err);
+  }
+}
+
+{
+  // Validate the gid option
+  if (!(common.isWindows && process.getgid() === 0)) {
+    const err = /TypeError: "gid" must be an integer/;
+
+    pass('gid', undefined);
+    pass('gid', null);
+    pass('gid', process.getgid());
+    fail('gid', __dirname, err);
+    fail('gid', true, err);
+    fail('gid', false, err);
+    fail('gid', [], err);
+    fail('gid', {}, err);
+  }
+}
+
+{
+  // Validate the shell option
+  const err = /TypeError: "shell" must be a boolean or string/;
+
+  pass('shell', undefined);
+  pass('shell', null);
+  pass('shell', false);
+  fail('shell', 0, err);
+  fail('shell', 1, err);
+  fail('shell', [], err);
+  fail('shell', {}, err);
+}
+
+{
+  // Validate the argv0 option
+  const err = /TypeError: "argv0" must be a string/;
+
+  pass('argv0', undefined);
+  pass('argv0', null);
+  pass('argv0', 'myArgv0');
+  fail('argv0', 0, err);
+  fail('argv0', 1, err);
+  fail('argv0', true, err);
+  fail('argv0', false, err);
+  fail('argv0', [], err);
+  fail('argv0', {}, err);
+}
+
+{
+  // Validate the windowsVerbatimArguments option
+  const err = /TypeError: "windowsVerbatimArguments" must be a boolean/;
+
+  pass('windowsVerbatimArguments', undefined);
+  pass('windowsVerbatimArguments', null);
+  pass('windowsVerbatimArguments', true);
+  pass('windowsVerbatimArguments', false);
+  fail('windowsVerbatimArguments', 0, err);
+  fail('windowsVerbatimArguments', 1, err);
+  fail('windowsVerbatimArguments', __dirname, err);
+  fail('windowsVerbatimArguments', [], err);
+  fail('windowsVerbatimArguments', {}, err);
+}
+
+{
+  // Validate the timeout option
+  const err = /TypeError: "timeout" must be a non-negative number/;
+
+  pass('timeout', undefined);
+  pass('timeout', null);
+  pass('timeout', 1);
+  pass('timeout', 0);
+  fail('timeout', -1, err);
+  fail('timeout', true, err);
+  fail('timeout', false, err);
+  fail('timeout', __dirname, err);
+  fail('timeout', [], err);
+  fail('timeout', {}, err);
+}
+
+{
+  // Validate the maxBuffer option
+  const err = /TypeError: "maxBuffer" must be an unsigned integer/;
+
+  pass('maxBuffer', undefined);
+  pass('maxBuffer', null);
+  pass('maxBuffer', 1);
+  pass('maxBuffer', 0);
+  fail('maxBuffer', 3.14, err);
+  fail('maxBuffer', -1, err);
+  fail('maxBuffer', NaN, err);
+  fail('maxBuffer', Infinity, err);
+  fail('maxBuffer', true, err);
+  fail('maxBuffer', false, err);
+  fail('maxBuffer', __dirname, err);
+  fail('maxBuffer', [], err);
+  fail('maxBuffer', {}, err);
+}
+
+{
+  // Validate the killSignal option
+  const typeErr = /TypeError: "killSignal" must be a string or number/;
+  const rangeErr = /RangeError: "killSignal" cannot be 0/;
+  const unknownSignalErr = /Error: Unknown signal/;
+
+  pass('killSignal', undefined);
+  pass('killSignal', null);
+  pass('killSignal', 'SIGKILL');
+  pass('killSignal', 500);
```

Comment:
I think the docs are wrong. It allows signal numbers from here - https://github.com/nodejs/node/blob/9e4a899d31181a34b20b99785fae7a6a97210e60/lib/child_process.js#L428.